### ISSUE TITLE
MAINT: Automatically release tagged versions on PyPI

### ISSUE THREAD

**Issue by larsoner** [mne-tools/mne-python#11509]
I've opened separate issues for the points that I think are the major pain points:

1. Versions, tagging: https://github.com/mne-tools/mne-python/issues/11466
2. Doc versioning: https://github.com/mne-tools/mne-python/issues/11507
3. Changelog maintenance: https://github.com/mne-tools/mne-python/issues/11508

I'm not going to open a separate issue for the remaining parts, but once this stuff is in place, ideally making a release on GitHub with a proper tag (e.g., `v1.4.0`) would automagically:

1. Be picked up as version number change in `setuptools_scm`
2. Be picked up as a new doc version to deploy by our docs infrastructure
3. Push a release to PyPI (this part [is trivial with GH actions nowadays](https://github.com/mne-tools/mne-bids-pipeline/blob/eea13e91f5ec27424e8453cf3f65f5148016e96e/.github/workflows/release.yml))
4. Be updated on conda-forge (this part is automatic)

Then the only manual step that remains would be to build + update installers. We could separately automate at least the building process probably by sending an event from mne-feedstock's conda-forge PR merge to tell the installers to kick off an update, but given that this usually requires a time delay for CDNs anyway, it's probably not worth it for now.

**Comment by larsoner** [comment]
I think for as long as I can remember, any tagged version on GitHub ended up being a PyPI release. So I think the low-hanging fruit here is to enable tagged releases like https://github.com/mne-tools/mne-bids-pipeline/blob/main/.github/workflows/release.yml but following whatever the latest best practice is (I think it has changed). I'm going to re-title the issue accordingly since we have #11507 and #11508 for the harder stuff already.
